app/assets/javascripts/components/edit_campaign_page.js.jsx

```diff
@@ -0,0 +1,94 @@
+CheckBox = React.createClass({
+  getInitialState() {
+    return (
+      { checked: this.props.checked }
+    )
+  },
+
+  handleChange() {
+    checked = React.findDOMNode(this.refs.checkbox).checked
+
+    data = {}
+    data[this.props.name] = checked
+    this.props.onChange(data);
+    this.setState( {checked: !this.state.checked } )
+  },
+
+  render() {
+    return (
+      <div className="checkbox">
+        <label htmlFor="campaign_page_{this.props.label}" className="control-label">
+          <input type="checkbox" checked={this.state.checked } onChange={ this.handleChange } ref="checkbox" /> {this.props.label}
+        </label>
+      </div>
```

Comment:
@NealJMD the mixing of presentation and logic is the 'react way'. 

With react, presentation and logic are part of the same concern. The framework shines when you create tiny reusable components. You build up your app by composing these units into a whole. It's the components that define your separation of concerns. Coming from an MVC background (Backbone with Marionette), react's approach at first felt jarring. I've now come to really like its design philosophy.